### PR TITLE
Call activator on off-heap container and remove entris from all stores

### PR DESCRIPTION
**Pull request by vjuranek**
https://issues.jboss.org/browse/ISPN-8442
https://issues.jboss.org/browse/ISPN-8443

### CODE REVIEW

**Comment by wburns** [comment]
Everything else looks good though :+1:

**Comment by vjuranek** [comment]
Added also into `put` method.

**Comment by wburns** [comment]
Changes look good to me, just want to wait on CI but then I will integrate. Thanks @vjuranek !

**Comment by galderz** [comment]
Needs rebase

**Comment by vjuranek** [comment]
rebased

**Comment by wburns** [comment]
Restarted build again, maybe this will get a finished build finally.

**Comment by wburns** [comment]
I would recommend rebasing to get fixes from master to better run CI.

**Comment by vjuranek** [comment]
rebased

**Comment by vjuranek** [comment]
test failures are unrealted

**Comment by wburns** [comment]
Integrated into 9.1.x and master, thanks @vjuranek !

File: core/src/main/java/org/infinispan/eviction/impl/ActivationManagerImpl.java [hunk 1]
```diff
@@ -88,7 +87,7 @@ public void onRemove(Object key, boolean newEntry) {
          } else {
             //the entry already exists in data container. It may be put during the load by the CacheLoaderInterceptor
             //so it was already activate in the private stores.
-            if (primaryOwner && persistenceManager.deleteFromAllStores(key, SHARED) && statisticsEnabled) {
+            if (primaryOwner && persistenceManager.deleteFromAllStores(key, BOTH) && statisticsEnabled) {
```
Comment (danberindei) [inline core/src/main/java/org/infinispan/eviction/impl/ActivationManagerImpl.java]: We should really stop trying to make passivation work with shared stores: https://issues.jboss.org/browse/ISPN-7168

**Pull request merged**